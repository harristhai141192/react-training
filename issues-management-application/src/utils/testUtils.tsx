// Libraries
import React from 'react';
import { ThemeProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

// Themes
import themes from '@themes/index';

const customRender = (component: React.ReactNode) => {
  return render(
    <ThemeProvider theme={themes}>
      <Router>{component}</Router>
    </ThemeProvider>,
  );
};

export * from '@testing-library/react';
export { customRender as render };
