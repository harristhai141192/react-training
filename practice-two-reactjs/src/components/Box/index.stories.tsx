// Libs
import { ComponentStory, ComponentMeta } from '@storybook/react';

// Component
import Box from '.';

import '../../index.css';

export default {
  title: 'Components/Box',
  component: Box,
  argTypes: {},
} as ComponentMeta<typeof Box>;

const Template: ComponentStory<typeof Box> = (args) => <Box {...args} />;

export const Default = Template.bind({});
Default.args = {
  imageURL: 'src/assets/images/photoAva1.png',
  subText:
    'Launch campaigns but also find new customers. Our unique platform handles campaigns from start to end.',
  maxWidth: '35%',
};
