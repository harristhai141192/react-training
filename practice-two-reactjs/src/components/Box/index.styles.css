.textDetail {
  display: flex;
  justify-content: left;
  flex-direction: column;
  align-items: flex-start;
  font-family: "RalewayLight";
  padding: 30px;
}

.cardImage {
  overflow: hidden;
  height: 220px;
}