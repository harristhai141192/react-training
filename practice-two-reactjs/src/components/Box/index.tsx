import { Box, Text, Image } from '@chakra-ui/react';

interface IProps {
  imageURL?: string;
  imageAlt?: string;
  subText?: string;
  maxWidth?: string;
  width?: { sm: string; md?: string; lg?: string };
}
// FOLDER NAME COMPONENT NOT MATCH
const BoxComponent = ({ imageURL, imageAlt, subText, maxWidth, width }: IProps) => {
  return (
    <Box
      display='flex'
      flexDirection='column'
      maxW={maxWidth}
      boxShadow='0 0 10px 1px lightgrey'
      overflow='hidden'
      width={width}
      bg='white'
    >
      <Image maxHeight='200px' objectFit='cover' src={imageURL} w='100%' alt={imageAlt} />
      <Box p={5} bg='white'>
        <Box display='flex' alignContent='flex-start'>
          <Text fontSize='sm' textAlign='left' fontFamily='RalewayLight'>
            {subText}
          </Text>
        </Box>
        <Box display='flex' alignContent='flex-start' fontFamily='RalewayBold'>
          <Text paddingTop={2} as='b' fontSize='sm' color='var(--main-color)'>
            Learn more {'>'}
          </Text>
        </Box>
      </Box>
    </Box>
  );
};

export default BoxComponent;
