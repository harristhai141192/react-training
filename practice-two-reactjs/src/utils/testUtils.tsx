import { ThemeProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';
import themes from '@themes/index';
import { ReactNode } from 'react';

const customRender = (component: ReactNode) => {
  return render(<ThemeProvider theme={themes}>{component}</ThemeProvider>);
};

export * from '@testing-library/react';
export { customRender as render };
