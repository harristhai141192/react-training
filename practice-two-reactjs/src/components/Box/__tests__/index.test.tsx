// libraries
import { getByAltText, getByRole, getByText, render } from '@utils/testUtils';
// components
import Box from '../index';

// theme

describe('Box component', () => {
  test('It should match snapshot', () => {
    const { container } = render(<Box />);

    expect(container).toMatchSnapshot();
  });

  test('Should render contain sub text, image and alt text', () => {
    const { container } = render(
      <Box
        imageAlt='for testing'
        imageURL='src/assets/images/photoAva1.png'
        subText='Launch campaigns but also find new customers. Our unique platform handles campaigns from start to end.'
        maxWidth='35%'
      />,
    );

    const altText = getByAltText(container, 'for testing');
    const subText = getByText(
      container,
      'Launch campaigns but also find new customers. Our unique platform handles campaigns from start to end.',
    );
    const role = getByRole(container, 'img');

    expect(altText).toBeTruthy;
    expect(subText).toBeTruthy;
    expect(role).toBeTruthy;
  });
});
