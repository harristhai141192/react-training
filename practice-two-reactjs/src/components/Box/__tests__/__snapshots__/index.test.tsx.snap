// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component It should match snapshot 1`] = `
<div>
  <div
    class="css-1xghscp"
  >
    <div
      class="cardImage"
    >
      <img
        class="chakra-image css-jus48v"
      />
    </div>
    <div
      class="textDetail"
    >
      <p
        class="chakra-text css-bbk83b"
      />
      <b
        class="chakra-text css-1ulx5am"
      >
        Learn more 
        <svg
          class="chakra-icon css-12pgzdw"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g
            fill="currentColor"
          >
            <path
              d="M13.584,12a2.643,2.643,0,0,1-.775,1.875L3.268,23.416a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L.768,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,13.584,12Z"
            />
            <path
              d="M23.75,12a2.643,2.643,0,0,1-.775,1.875l-9.541,9.541a1.768,1.768,0,0,1-2.5-2.5l8.739-8.739a.25.25,0,0,0,0-.354L10.934,3.084a1.768,1.768,0,0,1,2.5-2.5l9.541,9.541A2.643,2.643,0,0,1,23.75,12Z"
            />
          </g>
        </svg>
      </b>
    </div>
  </div>
</div>
`;
